pkg/engine/handlers/validation/validate_resource.go

```diff
@@ -170,16 +170,18 @@ func (v *validator) validate(ctx context.Context) *engineapi.RuleResponse {
 }
 
 func (v *validator) validateOldObject(ctx context.Context) (*engineapi.RuleResponse, error) {
-	pc := v.policyContext
-	oldPc, err := v.policyContext.OldPolicyContext()
+	var err error
+	v.policyContext, err = v.policyContext.OldPolicyContext()
 	if err != nil {
 		return nil, errors.Wrapf(err, "cannot get old policy context")
 	}
 
-	v.policyContext = oldPc
 	resp := v.validate(ctx)
-	v.policyContext = pc
 
+	v.policyContext, err = v.policyContext.RefreshPolicyContext()
```

Comment:
This may be confused with the context `Checkpoint` and `Restore` which seem similar but are different.